File: Third semester/Java/Exercises/Lab/Exercise7/src/com/shapes/Triangle.java
package com.shapes;

public class Triangle extends Shape {
    private double a;
    private double b;
    private double c;

    public Triangle(double a, double b, double c) {
        this.a = a;
        this.b = b;
        this.c = c;
    }

    @Override
    public double calculateArea() {
        double perimeter = calculatePerimeter() / 2;
        double area = Math.sqrt(perimeter * (perimeter - a) * (perimeter - b) * (perimeter - c));
        return area;
    }

    @Override
    public double calculatePerimeter() {
        return a + b + c;
    }
}

File: Fourth semester/Software Architectures/Exercises/Design Patterns/DesignPatterns/DecoratorExercise2/Triangle.cs
namespace DecoratorExercise2
{
    using System;

    public class Triangle : IShape
    {
        public Triangle(double sizeA, double sizeB, double sizeC)
        {
            this.SizeA = sizeA;
            this.SizeB = sizeB;
            this.SizeC = sizeC;
        }

        public double SizeA { get; set; }

        public double SizeB { get; set; }

        public double SizeC { get; set; }

        // Heron's Formula
        public void CalculateArea()
        {
            var halfPerimeter = (this.SizeA + this.SizeB + this.SizeC) / 2;
            var area = Math.Sqrt(halfPerimeter * (halfPerimeter - this.SizeA) * (halfPerimeter - this.SizeB) * (halfPerimeter - this.SizeC));
            Console.WriteLine();
            Console.Write($"{nameof(Triangle)} area is {area:F2}");
        }

        public void CalculatePerimeter()
        {
            var perimeter = this.SizeA + this.SizeB + this.SizeC;
            Console.WriteLine();
            Console.Write($"{nameof(Triangle)} perimeter is {perimeter:F2}");
        }

        public void Draw()
        {
            Console.WriteLine();
            Console.Write("Drawing triangle");
        }
    }
}
